Tests/SentryProfilerTests/SentryProfilerSwiftTests.swift

```diff
@@ -92,60 +97,48 @@ class SentryProfilerSwiftTests: XCTestCase {
 
             // gather mock cpu usages and memory footprints
             for _ in 0..<mockUsageReadingsPerBatch {
-                self.timerWrapper.fire()
+                self.metricTimerWrapper.fire()
             }
 
     #if !os(macOS)
-            var shouldRecordFrameRateExpectation = false
-            var previousSystemTime = currentDateProvider.systemTime()
+            var shouldRecordFrameRateExpectation = true
 
             func changeFrameRate(_ new: FrameRate) {
                 displayLinkWrapper.changeFrameRate(new)
                 shouldRecordFrameRateExpectation = true
             }
 
             func renderGPUFrame(_ type: GPUFrame) {
-                if shouldRecordFrameRateExpectation {
-                    shouldRecordFrameRateExpectation = false
-                    print("will expect frame rate \(displayLinkWrapper.currentFrameRate.rawValue) at \(previousSystemTime)")
-                    expectedFrameRateChanges.append([
-                        "elapsed_since_start_ns": String(previousSystemTime),
-                        "value": NSNumber(value: displayLinkWrapper.currentFrameRate.rawValue)
-                    ])
-                }
```

Comment:
The rest of the diff in this hunk deals with ordering of advancing the mocked system clock (which is now managed by the display link wrapper) since we record the "current" timestamp instead of "previous" now. Since some of the granularity of control was lost in this file by putting it there, we needed to reorder a few things. One of which was moving the creation of a refresh rate expectation until _after_ mocking a slow/frozen frame (which advances the system clock).